protocol.md

```diff
@@ -114,6 +114,114 @@ Observe the behavior of this algorithm in some failure scenarios. First, any of
+
+The user can pair with a helper in person, or it could be partially automated. The protocol assumes that one party gives the other contact information outside the protocol, which contains only the minimum information needed to start the protocol. The other party then sends a \"pairing request\", and receives back a \"pairing reply\". At that point, the two are paired.
+
+Partially automated pairing might happen between a customer and a company providing them an app. This might be a bank or credit union customer that gets the banking app from the bank. The customer might go through some kind of KYC (Know Your Customer) process where the bank verifies their identity before establishing an account for them. That process would end with them having an account and the app to access the account. It would also end up sending the app the contact information. and the user\'s app would then send the pairing request to the bank\'s server, which would reply. The contact information contains a random number which is the \"nonce\", which the user sends back to the bank as part of the pairing request. This allows the bank to ensure that the pairing request is coming from the same person who KYCed and established the account. In this scenario, the bank has servers that fill the role of helper.
+
+An alternative scenario is pairing in person.  The sharer has an app on their phone, and the helper might be a friend or family member that has an app on their phone. These can be unrelated apps, as long as both support the DeRec protocol. Each would launch their app and press a button to say that they want to pair with the other. The app might have two different buttons, to allow the user to pair as the sharer or as the helper. Or the app might have a simpler design, with a single pairing button, and then would actually run the protocol twice: once to pair as the sharer, and once to pair as the helper. That would allow two friends or family members to each be a helper to the other.
+
+For an in-person pairing, each phone might display a QR code containing the contact information. This just contains the user\'s encryption public key, internet communication information, and the nonce. If the pairing fails, then they can try again, and the QR code will have a different nonce. Either user could then use their phone camera to see the contact displayed by the other user. It doesn\'t matter whether the contact is being sent from sharer to helper or from helper to sharer. Either way, the protocol will allow them to pair.
```

Comment:
Where we don't use RSA I guess it's just a public key rather than 'encryption' public key, as you can't directly encrypt with ECC without using ElGamal or equiv.

Reply:
It's the public information needed for encryption. For something like ElGamal, we could call it something other than a "key", if there's a better term for it. For asymmetric cryptography, I use the terms public key and private key. But it's fine to use a different term if you want to use a different term that's more general.

But whatever term we use, we need to convey that it's the public (vs private) thing, and that it's the thing used for encryption (vs signing).

Reply:
OK let's leave as is.